rust/operator-binary/src/druid_connection_controller.rs

```diff
@@ -0,0 +1,316 @@
+                            &druid_connection.spec.superset_cluster_name,
+                            Some(&druid_connection.spec.superset_cluster_namespace),
+                        )
+                        .await
+                        .context(SupersetDBRetrieval)?
+                        .status;
+                    if let Some(s) = superset_db_status {
+                        s.condition == SupersetDBStatusCondition::Ready
+                    } else {
+                        false
+                    }
+                } else {
+                    false
+                };
+                // Is the referenced druid discovery configmap there?
+                let druid_discovery_cm_exists = client
+                    .exists::<ConfigMap>(
+                        &druid_connection.spec.druid_cluster_name,
+                        Some(&druid_connection.spec.druid_cluster_namespace),
+                    )
+                    .await
+                    .context(DruidDiscoveryCheck)?;
+
+                if superset_db_ready && druid_discovery_cm_exists {
+                    let superset_db = client
+                        .get::<SupersetDB>(
+                            &druid_connection.spec.superset_cluster_name,
+                            Some(&druid_connection.spec.superset_cluster_namespace),
+                        )
+                        .await
+                        .context(SupersetDBRetrieval)?;
+                    // Everything is there, retrieve all necessary info and start the job
+                    let sqlalchemy_str = get_sqlalchemy_uri_for_druid_cluster(
+                        &druid_connection.spec.druid_cluster_name,
+                        &druid_connection.spec.druid_cluster_namespace,
+                        client,
+                    )
+                    .await?;
+                    let job =
+                        build_import_job(&druid_connection, &superset_db, &sqlalchemy_str).await?;
+                    client
+                        .apply_patch(FIELD_MANAGER_SCOPE, &job, &job)
+                        .await
+                        .context(ApplyJob)?;
+                    // The job is started, update status to reflect new state
+                    client
+                        .apply_patch_status(FIELD_MANAGER_SCOPE, &druid_connection, &s.importing())
+                        .await
+                        .context(ApplyStatus)?;
+                }
+            }
+            DruidConnectionStatusCondition::Importing => {
+                let ns = druid_connection.metadata.namespace.clone().unwrap();
+                let job_name = druid_connection.job_name();
+                let job = client
+                    .get::<Job>(&job_name, Some(&ns))
+                    .await
+                    .context(GetImportJob {
+                        namespace: ns,
+                        name: job_name,
+                    })?;
```

Comment:
You can use ObjectRef<_>:

```
    #[snafu(display(
        "druid connection state is 'importing' but failed to find job {obj_ref}",
    ))]
    GetImportJob {
        source: stackable_operator::error::Error,
        obj_ref: ObjectRef<Job>,
    },
```
and call
`
.context(GetImportJob {
                       obj_ref: ObjectRef::new(&job_name).within(&ns)
                    })?;
`